### PR TITLE
Configure SAL4J to publish to maven central

### PR DESCRIPTION
**Pull request by adam-kimpton**
- Upgraded gradle wrapper to the latest version (8.10.1)
- Upgraded to java 17
- Added github workflows & dependabot
- Bumped gradle dependencies
- Configured publishing to maven central

### CODE REVIEW

**Comment by stephen-rouxel** [comment]
@adam-kimpton I cannot see any issues with these changes. Since this will subsequently require a merge to master (otherwise it won't push to Maven Central) I'll await approval from @al-jeyapal for this PR and in turn merging into master.

**Review by stephen-rouxel — APPROVED** [review]
@adam-kimpton I cannot see any issues with these changes. Since this will subsequently require a merge to master (otherwise it won't push to Maven Central) I'll await approval from @al-jeyapal for this PR and in turn merging into master.

File: src/main/java/com/brightsparklabs/sal4j/AuditEvent.java [hunk 1]
```diff
@@ -19,24 +19,24 @@
 @Value.Immutable
 @JsonDeserialize(builder = ImmutableAuditEvent.Builder.class)
 interface AuditEvent {
-    /** @return The action which was performed. */
+    /** Returns The action which was performed. */
```
Comment (al-jeyapal) [inline src/main/java/com/brightsparklabs/sal4j/AuditEvent.java:22]: We should leave the @return in as well as its correct javadoc, will put back in

File: README.md [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 
-[ ![Download](https://api.bintray.com/packages/brightsparklabs/maven/sal4j/images/download.svg) ](https://bintray.com/brightsparklabs/maven/sal4j/_latestVersion)
+[![Build Status](https://github.com/brightsparklabs/sal4j/actions/workflows/java.yml/badge.svg)](https://github.com/brightsparklabs/sal4j/actions/workflows/java.yml)
+[![Maven](https://img.shields.io/maven-central/v/com.brightsparklabs/sal4j)](https://search.maven.org/artifact/com.brightsparklabs/sal4j)
 
 A simple Audit Logging framework for Java which can produce structured logs in JSON format.
 
+# Prerequisites
+
+- Java 17 or higher.
```
Comment (al-jeyapal) [inline README.md:10]: Will add a compatibility matrix instead, like our other projects.

**Pull request merged**